api/client/webclient/webclient.go

```diff
@@ -88,7 +92,7 @@ func doWithFallback(clt *http.Client, allowPlainHTTP bool, req *http.Request) (*
 
 // Find fetches discovery data by connecting to the given web proxy address.
 // It is designed to fetch proxy public addresses without any inefficiencies.
-func Find(ctx context.Context, proxyAddr string, insecure bool, pool *x509.CertPool) (*PingResponse, error) {
+func Find(ctx context.Context, proxyAddr string, insecure bool, pool *x509.CertPool, extraHeaders map[string]string) (*PingResponse, error) {
```

Comment:
Instead of adding a 5th or 6th argument to each function here, how about we encapsulate them into a configuration struct?

Find, Ping, GetTunnelAddr and GetMOTD all accept pretty much the same parameters so we could do something like:

```go
type WebClientConfig struct {
  Context context.Context
  ProxyAddr string
  Insecure bool
  Pool *x509.CertPool
  ConnectorName string
  ExtraHeaders map[string]string
}

func Find(conf WebClientConfig) (*PingResponse, error) { ... }
func Ping(conf WebClientConfig) (...)
```

And then callers will only use fields they need.